port/osx/omrvmem.c

```diff
@@ -676,32 +676,42 @@ getMemoryInRange(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier
 			currentAddress = (void *)alignmentInBytes;
 		}
 	}
-
-	if (NULL == memoryPointer) {
-		/* Try all addresses within range */
-		while ((startAddress <= currentAddress) && (endAddress >= currentAddress)) {
-			memoryPointer = reserveMemory(portLibrary, currentAddress, byteAmount, identifier, mode, pageSize, pageFlags, category);
-
-			if (NULL != memoryPointer) {
-				/* stop if returned pointer is within range */
-				if ((startAddress <= memoryPointer) && (endAddress >= memoryPointer)) {
-					break;
-				}
-				if (0 != omrvmem_free_memory(portLibrary, memoryPointer, byteAmount, identifier)) {
-					return NULL;
-				}
-				memoryPointer = NULL;
+	
+#if defined(VM_FLAGS_SUPERPAGE_SIZE_2MB)
+	if (PPG_vmem_pageSize[2] == pageSize) {
+		/* Don't bother scanning the entire address space if the request can't be satisfied. */
+		void *tmpPointer = reserveMemory(portLibrary, NULL, byteAmount, identifier, mode, pageSize, pageFlags, category);
+		if (NULL == tmpPointer) {
+			goto done;
+		}
+		omrvmem_free_memory(portLibrary, tmpPointer, byteAmount, identifier);
+	}
+#endif /* defined(VM_FLAGS_SUPERPAGE_SIZE_2MB) */
+	
+	/* Try all addresses within range */
+	while ((startAddress <= currentAddress) && (endAddress >= currentAddress)) {
+		memoryPointer = reserveMemory(portLibrary, currentAddress, byteAmount, identifier, mode, pageSize, pageFlags, category);
+
+		if (NULL != memoryPointer) {
+			/* stop if returned pointer is within range */
+			if ((startAddress <= memoryPointer) && (endAddress >= memoryPointer)) {
+				break;
+			}
+			if (0 != omrvmem_free_memory(portLibrary, memoryPointer, byteAmount, identifier)) {
+				goto done;
```

Comment:
I think `memoryPointer` can be non-null before the `goto`. So, it should be set to `NULL` before the `goto` in order to match the old code.
```suggestion
				memoryPointer = NULL;
				goto done;
```

Reply:
Right, thanks. Fixed.